Decorator/TextDecorator/MessageDecorator.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Decorator
+{
+    public class MessageDecorator : IMessage
+    {
+        protected IMessage message;
```

Comment:
You should have access only to public methods from inherited interface

Reply:
done

Reply:
done